weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -1513,6 +1522,54 @@ def execute_batch_action(
 
         return tsi.ExecuteBatchActionRes()
 
+    def completions_create(
+        self, req: tsi.CompletionsCreateReq
+    ) -> tsi.CompletionsCreateRes:
+        model_name = req.model_name
+        model_info = self._model_to_provider_info_map.get(model_name)
+        if not model_info:
+            raise InvalidRequest(f"No model info found for model {model_name}")
+        secret_fetcher = _secret_fetcher_context.get()
+        if not secret_fetcher:
+            raise InvalidRequest(
+                f"No secret fetcher found, cannot fetch API key for model {model_name}"
+            )
+        secret_name = model_info.get("api_key_name")
+        if not secret_name:
+            raise InvalidRequest(f"No secret name found for model {model_name}")
+        api_key = secret_fetcher.fetch(secret_name).get("secrets", {}).get(secret_name)
+        if not api_key:
+            raise InvalidRequest(f"No API key found for model {model_name}")
+        start = tsi.StartedCallSchemaForInsert(
+            project_id=req.project_id,
+            wb_user_id=req.wb_user_id,
+            op_name="call_llm",
+            started_at=datetime.datetime.now(),
+            inputs={**req.inputs.model_dump(), "model_name": req.model_name},
+            attributes={},
+        )
+        ch_call = _start_call_for_insert_to_ch_insertable_start_call(start)
+        self._insert_call(ch_call)
+        res = lite_llm_completion(api_key, model_name, req.inputs)
+        end = tsi.EndedCallSchemaForInsert(
```

Comment:
nit: i would create a start time, run the lite_llm_completion, mark the end time, then do all the construction of the objects after that. your timing includes clickhouse inserts

Reply:
Also, you can insert both records in the same network request then with `_insert_call_batch`